packages/@o3r/eslint-config/schematics/ng-add/eslint/index.ts

```diff
@@ -1,29 +1,69 @@
 import * as path from 'node:path';
-import type {
-  JsonObject,
+import {
+  fragment,
+  type JsonObject,
 } from '@angular-devkit/core';
 import {
   apply,
+  applyToSubtree,
   chain,
   MergeStrategy,
   mergeWith,
+  noop,
   renameTemplateFiles,
   type Rule,
+  type SchematicContext,
   template,
+  type Tree,
   url,
 } from '@angular-devkit/schematics';
 import {
   updateOrAddTsconfigEslint,
 } from '../tsconfig/index';
 
+const editAngularJson = (projectName: string, extension: string): Rule => async (tree: Tree, context: SchematicContext) => {
+  const { getWorkspaceConfig } = await import('@o3r/schematics');
+  const workspace = getWorkspaceConfig(tree);
+  if (!workspace) {
+    return;
+  }
+  const workspaceProject = projectName ? workspace.projects[projectName] : undefined;
+
+  if (!workspaceProject) {
+    context.logger.warn('No project detected, linter task can not be added.');
+    return;
+  }
```

Comment:
```suggestion
  let workspace: WorkspaceSchema | null = null;
  try {
    const { getWorkspaceConfig } = await import('@o3r/schematics');
    workspace = getWorkspaceConfig(tree);
  } catch {
    context.logger.warn('No @o3r/schematics installed, the workspace detection will be skipped');
  }
  const workspaceProject = projectName ? workspace?.projects[projectName] : undefined;

  if (!workspaceProject) {
    context.logger.warn('No project detected, linter task can not be added.');
    return;
  }
```

to not throw and add the package even without @o3r/schematics

(same change in below function)

Reply:
as discussed not applicable below but fix for this one